core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckUpdateListener.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.healthcheck;
+
+/**
+ * A listener interface for receiving {@link HealthCheckUpdateHandler} events.
+ */
+@FunctionalInterface
+public interface HealthCheckUpdateListener {
+
+    /**
+     * Invoked when the healthiness is updated.
+     */
+    void onUpdate(HealthCheckUpdateResult result) throws Exception;
```

Comment:
- How about `healthUpdated(HealthCheckUpdateResult)`?
- What other parameters users will be interested? Do we need to add `ServiceRequestContext` or something else?

Reply:
> What other parameters will users be interested? Do we need to add ServiceRequestContext or something else?

If users want `ServiceRequestContext`, they can use implementation of `HealthCheckUpdateHandler`.

Reply:
Sounds good.